nullaway/src/main/java/com/uber/nullaway/handlers/OptionalEmptinessHandler.java

```diff
@@ -155,33 +181,81 @@ private void updateIfAssertIsPresentTrueOnOptional(
             MethodInvocationNode argMethod = (MethodInvocationNode) unwrappedArg;
             Symbol.MethodSymbol argSymbol = ASTHelpers.getSymbol(argMethod.getTree());
             if (optionalIsPresentCall(argSymbol, types)) {
-              updateNonNullAPsForElement(
-                  bothUpdates, getOptionalGetter(argSymbol), argMethod.getTarget().getReceiver());
+              updateNonNullAPsForOptionalContent(bothUpdates, argMethod.getTarget().getReceiver());
             }
           }
         }
       }
     }
   }
 
-  private void updateNonNullAPsForElement(
-      AccessPathNullnessPropagation.Updates updates, @Nullable Element elem, Node base) {
-    if (elem != null) {
-      AccessPath ap = AccessPath.fromBaseAndElement(base, elem);
-      if (ap != null) {
-        updates.set(ap, Nullness.NONNULL);
-      }
+  private void updateNonNullAPsForOptionalContent(
+      AccessPathNullnessPropagation.Updates updates, Node base) {
+    AccessPath ap = AccessPath.fromBaseAndElement(base, getOptionalContentElement());
+    if (ap != null && base.getTree() != null) {
+      optionalExpressionToAccessPathMap.put(base.getTree().toString(), ap);
```

Comment:
Why do we need this `optionalExpressionToAccessPathMap`?  Isn't it sufficient to just have the relevant `AccessPath` in the store?